Bug: Double Free on aborted write operation

### What happened?

When I run the appended benchmark script and stop the python process, e.g. by stopping the debugging in VS Code, the server crashes. It reports an aborted write operation and stops with double free error (see log). 

### Version

ReductStore 1.2.2 from Docker Hub pulled on 02.01.2023 as latest

### OS

Linux

### Snippet to repeat the bug

Simple Benchmark script derived from the example code:

```
import time
import asyncio
import datetime

import numpy as np
from reduct import Client, Bucket

api_token="benchmark-..."

num_files = 100000
file_size = int(2.5e6)

print_interval = 50

async def main():
    # Create a client for interacting with the ReductStore service
    client = Client("http://nas:8383", api_token=api_token)

    # Create a bucket and store a reference to it in the `bucket` variable
    bucket: Bucket = await client.create_bucket("testmachine", exist_ok=True)

    rand_img = np.random.randint(0,256, file_size, 'uint8')
    byte_img = rand_img.tobytes()

    start = datetime.datetime.now()
    last_curr = start

    for i in range(0, num_files):
        # Write data to the bucket
        ts = time.time_ns() / 1000
        await bucket.write(f"foo3_entry-{i}", byte_img, ts)

        if i != 0 and (i+1) % print_interval == 0:
            curr = datetime.datetime.now()

            usecs = (curr - last_curr) / datetime.timedelta(microseconds=1)
            speed = (print_interval / usecs)

            avg_usecs = (curr - start) / datetime.timedelta(microseconds=1)
            avg_speed = ((i+1) / avg_usecs)
            print(f"\rUploading with {speed*1000*1000:4.2f} files/s, done {i} uploads with {avg_speed*1000*1000:4.2f} files/s -> {avg_speed*file_size:4.2f} MB/s",end="")
            last_curr = curr


# Run the main function
asyncio.run(main())
```

### Logs

```Shell
2023-01-02 14:09:52.876 (64512)  [INFO] -- main.cc:41 ReductStore 1.2.2
2023-01-02 14:09:52.876 (64512)  [INFO] -- main.cc:55 Configuration: 
 	RS_LOG_LEVEL = INFO (default)
	RS_HOST = 0.0.0.0 (default)
	RS_PORT = 8383 (default)
	RS_API_BASE_PATH = / (default)
	RS_DATA_PATH = /data (default)
2023-01-02 14:09:54.729 (64512)  [INFO] -- storage.cc:43 Load 1 buckets
2023-01-02 14:09:54.729 (64512) [WARNING] -- token_auth.cc:61 API token is empty. No authentication.
2023-01-02 14:09:54.729 (64512)  [INFO] -- http_server.cc:388 Run HTTP server on http://0.0.0.0:8383/
2023-01-02 15:52:10.853 (64512) [ERROR] -- http_server.cc:90 Aborted write operation
free(): double free detected in tcache 2
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,11 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Crashing when post request is aborted by client, [PR-223](https://github.com/reductstore/reductstore/pull/223)
+
 ## [1.2.2] - 2022-12-20
 
 ### Fixed
 
-- Fix token validation for anonymous access, [PR-217](https://github.com/reductstore/reductstore/pull/217)
+- Token validation for anonymous access, [PR-217](https://github.com/reductstore/reductstore/pull/217)
 
 ## [1.2.1] - 2022-12-19
 

--- a/src/reduct/api/http_server.cc
+++ b/src/reduct/api/http_server.cc
@@ -88,7 +88,7 @@ class HttpServer : public IHttpServer {
 
       res->onAborted([this] {
         LOG_ERROR("Aborted write operation");
-        error_ = core::Error{.code = 400};
+        error_ = core::Error::BadRequest("Aborted write operation");
       });
     }
 
@@ -96,7 +96,6 @@ class HttpServer : public IHttpServer {
 
     void await_suspend(std::coroutine_handle<> h) const noexcept {
       if (finish_ || error_) {
-        res_->onData([](auto, bool) {});
         h.resume();
       } else {
         async::ILoop::loop().Defer([this, h] { await_suspend(h); });
@@ -140,7 +139,7 @@ class HttpServer : public IHttpServer {
     };
 
     auto SendError = [ctx, &method, &url, CommonHeaders](const core::Error &err) {
-      if (err.code >= 500) {
+      if (err.code >= Error::kInternalError) {
         LOG_ERROR("{} {}: {}", method, url, err.ToString());
       } else {
         LOG_DEBUG("{} {}: {}", method, url, err.ToString());

